ec2-metadata writes error messages to stdout

In cases where IMDS is disabled or otherwise unreachable, `ec2-metadata` prints an error messages beginning with `[ERROR] Could not get IMDSv2 token.` to stdout. This should go to stderr instead in order to avoid causing confusion for tools trying to consume the output of this command.

--- a/amazon-ec2-utils.spec
+++ b/amazon-ec2-utils.spec
@@ -1,6 +1,6 @@
 Name:      amazon-ec2-utils
 Summary:   A set of tools for running in EC2
-Version:   2.1.0
+Version:   2.2.0
 Release:   1%{?dist}
 License:   MIT
 Group:     System Tools
@@ -75,6 +75,9 @@ rm -rf $RPM_BUILD_ROOT
 /etc/udev/rules.d/60-cdrom_id.rules
 
 %changelog
+* Mon Jun  5 2023 Guillaume Delacour <delacoug@amazon.com> - 2.2.0-1
+- Add `--quiet` option to `ec2-metadata --help` output
+- Add `-R`/`--region` option to `ec2-metadata` to discover the EC2 instance's region
 
 * Thu Apr  6 2023 Noah Meyerhans <nmeyerha@amazon.com> - 2.1.0-1
 - Add --quiet option to ec2-metadata

--- a/amazon-ec2-utils.spec
+++ b/amazon-ec2-utils.spec
@@ -75,6 +75,9 @@ rm -rf $RPM_BUILD_ROOT
 /etc/udev/rules.d/60-cdrom_id.rules
 
 %changelog
+* Thu Jan 18 2024 Keith Gable <gablk@amazon.com> - 2.2.0-1
+- Corrected issue where an ec2-metadata error was written to stdout
+
 * Mon Jun  5 2023 Guillaume Delacour <delacoug@amazon.com> - 2.2.0-1
 - Add `--quiet` option to `ec2-metadata --help` output
 - Add `-R`/`--region` option to `ec2-metadata` to discover the EC2 instance's region

--- a/ec2-metadata
+++ b/ec2-metadata
@@ -8,7 +8,7 @@
 
 function print_help()
 {
-echo "ec2-metadata v0.1.3
+echo "ec2-metadata v0.1.4
 Use to retrieve EC2 instance metadata from within a running EC2 instance. 
 e.g. to retrieve instance id: ec2-metadata -i
 		 to retrieve ami id: ec2-metadata -a
@@ -53,7 +53,7 @@ function set_imds_token()
 	if [ -z "${IMDS_TOKEN}" ];then
 		IMDS_TOKEN=$(curl -s -f -X PUT -H "X-aws-ec2-metadata-token-ttl-seconds: 900" ${METADATA_BASEURL}/${METADATA_TOKEN_PATH})
 		if [ "${?}" -gt 0 ] || [ -z "${IMDS_TOKEN}" ]; then
-			echo '[ERROR] Could not get IMDSv2 token. Instance Metadata might have been disabled or this is not an EC2 instance.'
+			echo '[ERROR] Could not get IMDSv2 token. Instance Metadata might have been disabled or this is not an EC2 instance.' >&2
 			exit 1
 		fi
 	fi

